src/simulation/LoadGenerator.cpp

```diff
@@ -421,7 +376,92 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
         return;
     }
 
-    // First time calling tx load generation
+    if (!mLoadTimer)
+    {
+        mLoadTimer = std::make_unique<VirtualTimer>(mApp.getClock());
+    }
+
+    if (mApp.getState() == Application::APP_SYNCED_STATE)
+    {
+        mLoadTimer->expires_from_now(std::chrono::milliseconds(STEP_MSECS));
+        mLoadTimer->async_wait([this, cfg]() { this->generateLoad(cfg); },
+                               &VirtualTimer::onFailureNoop);
+    }
+    else
+    {
+        CLOG_WARNING(
+            LoadGen,
+            "Application is not in sync, load generation inhibited. State {}",
+            mApp.getStateHuman());
+        mLoadTimer->expires_from_now(std::chrono::seconds(10));
+        mLoadTimer->async_wait(
+            [this, cfg]() { this->scheduleLoadGeneration(cfg); },
+            &VirtualTimer::onFailureNoop);
+    }
+}
+
+bool
+GeneratedLoadConfig::isDone() const
+{
+    return (isCreate() && nAccounts == 0) || (isLoad() && nTxs == 0) ||
+           (isSorobanSetup() && nInstances == 0);
+}
+
+bool
+GeneratedLoadConfig::areTxsRemaining() const
+{
+    return (isCreate() && nAccounts != 0) || (!isCreate() && nTxs != 0);
+}
+
+void
+LoadGenerator::start(GeneratedLoadConfig& cfg)
+{
+    // Setup config for soroban modes
+    if (!mStarted && cfg.isSoroban())
```

Comment:
`start` should be a no-op except on the first invocation, so why not add `if (mStarted) return;` at the beginning and get rid of the checks in the individual ifs. Currently the only functions that get called multiple times in `start` are `createRootAccount` and `updateMinBalance`. If we need to call those periodically they probably don't belong in `start` (but I don't think we do).